src/block/BigDripleaf.php

```diff
@@ -0,0 +1,233 @@
+		$this->tilt = DripleafTiltType::NONE();
+		parent::__construct($idInfo, $name, $typeInfo);
+	}
+
+	protected function describeBlockOnlyState(RuntimeDataDescriber $w) : void{
+		$w->horizontalFacing($this->facing);
+		$w->dripleafTiltType($this->tilt);
+		$w->bool($this->head);
+	}
+
+	/** @return $this */
+	public function setTilt(DripleafTiltType $tilt) : self{
+		$this->tilt = $tilt;
+		return $this;
+	}
+
+	public function getTilt() : DripleafTiltType{
+		return $this->tilt;
+	}
+
+	/** @return $this */
+	public function setHead(bool $head) : self{
+		$this->head = $head;
+		return $this;
+	}
+
+	public function isHead() : bool{
+		return $this->head;
+	}
+
+	private function canBeSupportedBy(Block $block) : bool{
+		//TODO: Moss block
+		return
+			$block->hasSameTypeId($this) ||
+			$block->getTypeId() === BlockTypeIds::CLAY ||
+			$block->hasTypeTag(BlockTypeTags::DIRT) ||
+			$block->hasTypeTag(BlockTypeTags::MUD);
+	}
+
+	public function onNearbyBlockChange() : void{
+		if(
+			!$this->head && !$this->getSide(Facing::UP)->hasSameTypeId($this) ||
+			!$this->canBeSupportedBy($this->getSide(Facing::DOWN))
+		){
+			$this->position->getWorld()->useBreakOn($this->position);
+		}
+	}
+
+	public function place(BlockTransaction $tx, Item $item, Block $blockReplace, Block $blockClicked, int $face, Vector3 $clickVector, ?Player $player = null) : bool{
+		$block = $blockReplace->getSide(Facing::DOWN);
+		if(!$this->canBeSupportedBy($block)){
+			return false;
+		}
+		if($player !== null){
+			$this->facing = Facing::opposite($player->getHorizontalFacing());
+		}
+		if($block instanceof BigDripleaf && $block->hasSameTypeId($this)){
+			$this->facing = $block->getFacing();
+			$tx->addBlock($block->getPosition(), (clone $block)->setHead(false));
+		}
+		$this->head = true;
+		return parent::place($tx, $item, $blockReplace, $blockClicked, $face, $clickVector, $player);
+	}
+
+	public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player $player = null, array &$returnedItems = []) : bool{
+		if($item instanceof Fertilizer && $this->grow($player)){
+			$item->pop();
+			return true;
+		}
+		return false;
+	}
+
+	private function seekToHead() : ?BigDripleaf{
+		if($this->head){
+			return $this;
+		}
+		$step = 1;
+		while(($next = $this->getSide(Facing::UP, $step)) instanceof BigDripleaf && $next->hasSameTypeId($this)){
+			if($next->isHead()){
+				return $next;
+			}
+			$step++;
+		}
+		return null;
+	}
+
+	private function grow(?Player $player) : bool{
+		$head = $this->seekToHead();
+		if($head === null){
+			return false;
+		}
+		$pos = $head->getPosition();
+		$up = $pos->up();
+		$world = $pos->getWorld();
+		if(
+			!$world->isInWorld($up->getFloorX(), $up->getFloorY(), $up->getFloorZ()) ||
+			$world->getBlock($up)->getTypeId() !== BlockTypeIds::AIR
+		){
+			return false;
+		}
+
+		$tx = new BlockTransaction($world);
+
+		$tx->addBlock($pos, (clone $head)->setHead(false));
```

Comment:
what about the tilt?

I'm getting the impression that BIG_DRIPLEAF_STEM ought to be separated from BIG_DRIPLEAF, since the tilt property does not apply to the stem.